### ISSUE TITLE
Statemachine Behaviour support

### ISSUE THREAD

**Issue by Seneral** [Baste-RainGames/Node_Editor#34]
**Description:** Integration into the existing calculation system, enables for transitioning though nodes based on conditions like the mecanim statemachine

**Progress:** With #e32476e, the WIP state system has been added. Transitioning behaviour exists, missing are the conditions. This is because any kind of functions cannot be serialized. Currently, I'm creating a workaroung called UnityFunc. It should be done and avaiable soon:)

**Comment by Barsonax** [comment]
For the conditions wouldnt it be nice if you could use nodes for this? Kinda like a decision tree?

On top of this a hierachical statemachine would be very useful and make this a pretty complete solution

**Comment by Seneral** [comment]
True, but it would be the same idea in the end. But I have not made progress here for a long time. Not that it would be impossible, but I had been focussing on other areas. I'd love to finish this stuff, it has been a hell lot of fun developing it. Also, I recently made a 'breakthrough' of serializing all kinds of actions for a different tool of mine, UndoPro, which would greatly help here.
But still I do not know when I will be able to work on it. I've near to zero time right now, with a vacation job and a main non-software project to do right now:(

**Comment by Barsonax** [comment]
Serializing actions and funcs would be very useful for implementing a statemachine (and for other stuff aswell).

Doing #70 first though since this kinda depends on it.

**Comment by Seneral** [comment]
Now that #70 has been implemented, I'm looking forward to update old works on the statemaschine feature to the modular way and take a look at improving it further:) Will post as soon as I have it ported.
